ner_v2/tests/test_number_detection.py

```diff
@@ -0,0 +1,223 @@
+        }
+        detect_texts, original_texts = get_number_from_number_word(message, number_word_map)
+        zipped = zip(detect_texts, original_texts)
+        self.assertEqual(len(zipped), 3)
+        self.assertIn((1001, 'one thousand one'), zipped)
+        self.assertIn((2, 'two'), zipped)
+        self.assertIn((3, 'three'), zipped)
+
+    def test_get_number_with_multiple_spaces_in_unit_scale_number_word(self):
+        """
+        Number detection from word with multiple spaces in unit scale number like 'one thousand   one   hundred two'
+        """
+        message = 'there are one thousand   one   hundred two students attending placement drive'
+        number_word_map = {
+            'one': NumberVariant(scale=1, increment=1),
+            'two': NumberVariant(scale=1, increment=2),
+            'hundred': NumberVariant(scale=100, increment=0),
+            'thousand': NumberVariant(scale=1000, increment=0)
+        }
+        detect_texts, original_texts = get_number_from_number_word(message, number_word_map)
+        zipped = zip(detect_texts, original_texts)
+        self.assertEqual(len(zipped), 1)
+        self.assertIn((1102, 'one thousand   one   hundred two'), zipped)
+
+    def test_en_number_detection_for_integer_number(self):
+        """
+        Number detection for english language for integer number like '100', '2'
+        """
+        message = u'100 got selected for interview'
+        number_detector_object = NumberDetector(entity_name=self.entity_name, language='en')
+        number_dicts, original_texts = number_detector_object.detect_entity(message)
+        zipped = zip(number_dicts, original_texts)
+        self.assertEqual(len(zipped), 1)
+        self.assertIn(({'value': '100', 'unit': None}, u'100'), zipped)
+
+    def test_en_number_detection_for_integer_number_with_unit(self):
+        """
+        Number detection for english language for integer number with units like 'Rs100', '2Rs'
+        """
+        message = u'rs.100 is the application charger'
+        number_detector_object = NumberDetector(entity_name=self.entity_name, language='en')
+        number_dicts, original_texts = number_detector_object.detect_entity(message)
+        zipped = zip(number_dicts, original_texts)
+        self.assertEqual(len(zipped), 1)
+        self.assertIn(({'value': '100', 'unit': 'rupees'}, 'rs.100'), zipped)
+
+    def test_en_number_detection_for_decimal_number(self):
+        """
+        Number detection for english language for decimal number like '100.2'
+        """
+        message = u'Todays temperature is 11.2 degree celsius'
+        number_detector_object = NumberDetector(entity_name=self.entity_name, language='en')
+        number_dicts, original_texts = number_detector_object.detect_entity(message)
+
+        zipped = zip(number_dicts, original_texts)
+        self.assertEqual(len(zipped), 1)
+        self.assertIn(({'value': '11.2', 'unit': None}, u'11.2'), zipped)
```

Comment:
This might need to change once someone adds temperature units

Reply:
Noted